Changes - edits (#232)
bcgov/platform-developer-docs: src/docs/automation-and-resiliency/request-quota-increase-for-openshift-project-set.md

---
# Request a quota increase for an OpenShift project set
Last updated: **March 26, 2024**

This guide shows you how to request more resources for your OpenShift project set on the [Platform Product Registry](https://registry.developer.gov.bc.ca/).

## On this page
* **[Prerequisites](#prerequisites)**
* **[Steps to request quota increase](#steps-to-request-quota-increase)**
* **[Essential resources to improve your quota increase request](#essential-resources-to-improve-your-quota-increase-request)**
* **[Related pages](#related-pages)**

## Prerequisites 
Before initiating the request, make sure you have completed the following steps:

1. **Evaluate resource usage:** Analyze your project's resource consumption to determine if a quota increase is necessary. Utilize tools like the OpenShift Dashboard or the command-line interface (CLI) for resource monitoring.

Use the following information to help determine the level of resource the application needs:

- [Resource management guidelines](https://github.com/BCDevOps/developer-experience/blob/master/docs/ResourceManagementGuidelines.md)
- [Application resource tuning](../automation-and-resiliency/application-resource-tuning.md)

> 2. **Familiarize yourself with Sysdig (optional):** Setting up Sysdig can provide comprehensive insights into your project's performance and resource utilization. However, it is not mandatory for requesting a quota increase and you can use a different system to provide the information the Platform Services Team requires to process your request increase.  If you are using it we provide more information in the section 

Reviewer: ```suggestion
2. **Familiarize yourself with Sysdig (optional):** Setting up Sysdig can provide comprehensive insights into your project's performance and resource utilization. However, it is not mandatory for requesting a quota increase and you can use a different system to provide the information the Platform Services Team requires to process your request increase.  If you are using it we provide more information in the section [Onboarding to application monitoring with Sysdig](../app-monitoring/sysdig-monitor-onboarding.md).
```

3. **Get in touch with the Platform Services Team**
This process suggests good communication from your team and the Platform Services Team particularly if you have questions prior your quota increase request, please reach out to them at [PlatformServicesTeam@gov.bc.ca](mailto:PlatformServicesTeam@gov.bc.ca) and enter  subject: Quota request with your inquiry.

## Steps to request quota increase

> 

Reviewer 2: ```suggestion
As part of our ongoing efforts to optimize resource allocation across our OpenShift environments, we request teams seeking a quota increase to provide detailed justifications. This structured approach ensures that all requests are evaluated fairly and resources are used efficiently.
```

### Step 1: Access the Platform Product Registry 

> Go to the [Platform Product Registry](https://registry.developer.gov.bc.ca/). Through dashboard of the product you wish to increase the quota. 

Author: ```suggestion
Access the [Platform Product Registry](https://registry.developer.gov.bc.ca/) and use the dashboard of your chosen product to adjust the quota.
```

### Step 2: Identify page and potential changes for each section

> The product page has different sections: 

Reviewer: does registry support changes in multiple sections during resource request?

Author: Yeah when testing, one has also the possibility to changing product owner's and technical lead's info

For more information on quotas, see [OpenShift project resource quotas](../automation-and-resiliency/openshift-project-resource-quotas.md).

Before we move forward with your quota change request, expect to discuss with the Platform Team ways to optimize your application within your current quota. We usually don't approve quota requests without trying to make the existing quota work first. The more details you provide in your initial request, the shorter this conversation is likely to be!

If you need help answering this question [check these examples](#examples-of-answers).

### Step 5: Submit edit request changes 

The Platform Services team needs to know if your application is using the current quotas efficiently. 

* Once all the information has been compiled from [**Step 4**](#step-4-gather-information-for-your-request) click on the `Submit edit request` button

* A pop-up will appear to allow space to include all the information you have collected and gathered, this place only allows for text information if you need to submit photos or files please do so in the next step as it is necessary and part of the request process

> ### Step 6: Email the Platform Services team and book a meeting with them

Reviewer 3: ```suggestion
### Step 6: Email the Platform Services team
```

The last step is to email the request to the Platform Services team at [PlatformServicesTeam@gov.bc.ca](mailto:PlatformServicesTeam@gov.bc.ca) and include the following information: 

* Attach screenshot of Sysdig dashboard (if available)
* Attach additional images or files that supplement the already submitted information in the previous step

> * Book a meeting with the Platform Services team

Reviewer 3: We don't typically require a meeting for every quota request as a rule. I think we should remove this line - Billy will request a meeting if he needs to follow up in a discussion.

  - Expected quota increase amount with a detailed allocation plan

* **Justification for increase:** Clearly state the reasons for requesting a quota increase, such as anticipated growth, bursty workload patterns, or specific project requirements.

* **Impact Assessment:** Assess the impact of the increased quota on your project's performance, scalability, and budget.

If you need help answering this question [check these examples](#examples-of-answers).

#### 2. Can you tell us more about the current and desired states of the relevant Openshift objects?

Please provide information about how much resources the important pods are using right now.  If you can, also let us know how their resource usage will change if we increase the quota. You don't need to give us details about every single pod or deployment in your namespace. Just focus on the ones that matter for why you're asking for this increase.

 **Collect your application metrics:** Compare the current resource utilization and compare it with the quota. Specify your project's current resource quotas and usage.

>   - Current quota in the project set
>   - Total CPU, memory or storage currently used and any other resources relevant to your application's requirements.
> - Present the resource monitors within your namespace in order to understand  the level of resource consumption in your application and how efficiently it utilizes resources
> - Sysdig offers a pre-built dashboard template designed to visualize and analyze metrics in your OpenShift project. To simplify the process of requesting a quota increase, consider utilizing the Sysdig dashboard template. Instructions can be found in [here](../app-monitoring/sysdig-monitor-create-monitoring-dashboards.md)

Reviewer: ```suggestion
  - Current quota in the project set
  - Total CPU, memory or storage currently used and any other resources relevant to your application's requirements
  - Present the resource monitors within your namespace in order to understand  the level of resource consumption in your application and how efficiently it utilizes resources
  - Sysdig offers a pre-built dashboard template designed to visualize and analyze metrics in your OpenShift project. To simplify the process of requesting a quota increase, consider utilizing the Sysdig dashboard template. Instructions can be found in [here](../app-monitoring/sysdig-monitor-create-monitoring-dashboards.md)
```

### Step 6: Email the Platform Services team

The last step is to email the request to the Platform Services team at [PlatformServicesTeam@gov.bc.ca](mailto:PlatformServicesTeam@gov.bc.ca) and include the following information: 

* Attach screenshot of Sysdig dashboard (if available)
* Attach additional images or files that supplement the already submitted information in the previous step
* If you have further questions, it is also recommended to book a meeting with the Platform Services Team

## Essential resources to improve your quota increase request 

### Examples of requests
These examples mainly concentrate on particular pods or deployments. When you're asking for a quota increase, it's typically to support a specific deployment or group of deployments. Focus your request explanations on these - there's no need to spend time outlining the resource allocations for every single pod in your namespace, just provide the information that's relevant to the request.

> If you're unsure, it's better to include more information than less!

Reviewer 2: ```suggestion
If you're unsure, it's better to include more information than less, the Platform Team is here to help!
```
